backend/sites/src/app/services/site/site.service.ts

```diff
@@ -1116,4 +1129,540 @@ export class SiteService {
+      this.sitesLogger.debug(
+        'SiteService.getSiteDetailsPendingSRApproval() end',
+      );
+
+      return resultDTO;
+    } catch (error) {
+      this.sitesLogger.log(
+        'SiteService.getSiteDetailsPendingSRApproval() error' +
+          JSON.stringify(error),
+      );
+
+      throw new Error('Failed to determine banner type.');
+    }
+  }
+
+  /**
+   * Bulk Approval/ Reject For Site Registry Approvals
+   * @param inputDTO list of site
+   * @param userInfo authenticated user
+   * @returns true/false
+   */
+  async bulkUpdateForSR(
+    inputDTO: BulkApproveRejectChangesDTO,
+    userInfo: any,
+  ): Promise<boolean> {
+    try {
+      this.sitesLogger.log('SiteService.bulkUpdateForSR() start');
+      this.sitesLogger.debug('SiteService.bulkUpdateForSR() start');
+
+      if (!inputDTO) {
+        return false;
+      } else {
+        const { isApproved, sites } = inputDTO;
+
+        sites.forEach(async (site: SiteRecordsForSRAction) => {
+        await this.entityManager.transaction(
+          async (transactionalEntityManager: EntityManager) => {            
+              if (
+                (site != null && site.siteId === null) ||
+                site.siteId === undefined
+              ) {
+                return false;
+              }
+              await this.processSRBulkUpdates(
+                transactionalEntityManager,
+                site,
+                isApproved,
+                userInfo,
+              );
+            });
+          },
+        );
+
+        return true;
+      }
+    } catch (error) {
+      this.sitesLogger.log(
+        'SiteService.bulkUpdateForSR() error' + JSON.stringify(error),
+      );
+      return false;
+    }
+  }
+
+  async processSRBulkUpdates(
+    transactionalEntityManager: EntityManager,
+    site: SiteRecordsForSRAction,
+    isApproved: boolean,
+    userInfo: any,
+  ) {
+    try {
+      this.sitesLogger.log('SiteService.processSRBulkUpdates() start');
+
+      if ((site != null && site.siteId === null) || site.siteId === undefined) {
+        return false;
+      }
+
+      if (site.changes.indexOf('summary') !== -1) {
+        const sitesForUpdates = await transactionalEntityManager.find(Sites, {
+          where: { id: site.siteId, whoUpdated: site.whoUpdated },
+        });
+
+        if (sitesForUpdates.length > 0) {
+          sitesForUpdates.forEach((site) => {
+            this.setUpdatedStatus(site, isApproved, userInfo);
+          });
+          await transactionalEntityManager.save(Sites, sitesForUpdates);
+        }
+        else {
+          this.sitesLogger.log(
+            'SiteService.processSRBulkUpdates() no summary to process.',
+          );
+        }
+      }
+
+      if (site.changes.indexOf('notation') !== -1) {
+        const events = await transactionalEntityManager.find(Events, {
+          where: { siteId: site.siteId, whoUpdated: site.whoUpdated },
+        });
+
+        if (events.length > 0) {
```

Comment:
can events be null or empty ?

Reply:
it will not be null,     find<Entity extends ObjectLiteral>(entityClass: EntityTarget<Entity>, options?: FindManyOptions<Entity>): Promise<Entity[]>; returns an array always